tests/integration/test_actor_api_helpers.py

```diff
@@ -330,21 +329,26 @@ async def main() -> None:
             async with Actor:
                 print('Starting...')
                 input = await Actor.get_input() or {}
-                counter_key = cast(str, input.get('counter_key'))
-                cnt = await Actor.get_value(counter_key, 0)
+                counter_key = input.get('counter_key')
+                cnt = await Actor.get_value(counter_key, 0)  # type: ignore
 
                 if cnt < 2:
                     print(f'Rebooting (cnt = {cnt})...')
-                    await Actor.set_value(counter_key, cnt + 1)
+                    await Actor.set_value(counter_key, cnt + 1)  # type: ignore
                     await Actor.reboot()
                     await Actor.set_value('THIS_KEY_SHOULD_NOT_BE_WRITTEN', 'XXX')
 
-                this_should_not_be_written = await Actor.get_value('THIS_KEY_SHOULD_NOT_BE_WRITTEN')
-                assert this_should_not_be_written is None
                 print('Finishing...')
 
         actor = await make_actor('actor_rebooter', main_func=main)
         run_result = await actor.call(run_input={'counter_key': 'reboot_counter'})
+
+        not_written_value = await actor.last_run().key_value_store().get_record('THIS_KEY_SHOULD_NOT_BE_WRITTEN')
+        assert not_written_value is None
+
+        reboot_counter = await actor.last_run().key_value_store().get_record('reboot_counter')
+        assert reboot_counter['value'] == 2  # type: ignore
```

Comment:
Here the `assert reboot_counter is not None` would make for better error messages in case there are actually some issues with the reboot counter

Reply:
yeah, sure